pfio/v2/pathlib.py

```
@@ -1,271 +1,756 @@
 
-            while (i := parent.rfind(self.sep)) != -1:
-                parent = parent[:i]
-                if parent in visited_prefixes:
-                    continue
-                visited_prefixes.add(parent)
 
-                target_parts = (base / parent)._parts
+class Path(PurePath):
+    """
+    `pathlib.PosixPath` compatible interface.
 
-                if _test_glob_by_parts(target_parts, pattern_parts):
-                    yield self / parent
+    Args:
+        args: construct paths.
+        fs: target file system.
+        scheme: specify URL scheme. (for `as_uri` method)
 
-            target_parts = base_parts + p.split("/")
+    Note:
+        many methods raise `NotImplementedError`
+        because they require unsupported features of `FS`.
 
-            if _test_glob_by_parts(target_parts, pattern_parts):
-                yield self / p
+        several methods behave slightly different,
+        - `stat` returns `FileStat` object instead of `os.stat_result`.
+        - `glob`, `rglob`, `iterdir` will not return directory type object.
+    """
 
-    def stat(self):
-        return self._fs.stat(self.resolve())
+    def __init__(
+        self,
+        *args: str,
+        fs: FS,
+        scheme: Optional[str] = None,
+    ) -> None:
+        super().__init__(*args, fs=fs, scheme=scheme)
 
-    def unlink(self, missing_ok=False):
-        return self._fs.remove(self.resolve())
+    def _as_relative_to_fs(self) -> str:
+        return _removeprefix(self.as_posix(), self.anchor)
 
-    def rename(self, target):
-        raise NotImplementedError("rename() is unsupported on this system")
+    # ---------------------------------------
+    # pathlib.Path compatible classmethods
+    # ---------------------------------------
 
-    def read_text(self):
-        with self.open() as fp:
-            return fp.read()
+    @classmethod
+    def cwd(cls: Type[SelfPathType]) -> SelfPathType:
+        raise _not_supported()
```

Some FSes support .cwd including `pfio.v2.Local`. I want underlying FSes handle this method call (if it doesn't implement cwd, then let it throw unsupported exception. Oh, it has been never implemented. Nevermind..